linera-core/src/client/mod.rs

```diff
@@ -2030,17 +2040,22 @@ where
         };
         // Collect the hashed certificate values required for execution.
         let committee = self.local_committee().await?;
-        let blobs = self.read_local_blobs(block.published_blob_ids()).await?;
+        let published_blobs = self.read_local_blobs(block.published_blob_ids()).await?;
         // Create the final block proposal.
         let key_pair = self.key_pair().await?;
         let proposal = if let Some(cert) = manager.requested_locked {
-            Box::new(BlockProposal::new_retry(round, *cert, &key_pair, blobs))
+            Box::new(BlockProposal::new_retry(
+                round,
+                *cert,
+                &key_pair,
+                manager.locked_published_blobs,
```

Comment:
I see, we're already doing that anyway! Then the change to `read_local_blobs` isn't necessary, and the call should move into the `else` branch here, shouldn't it?